### DISCUSSION TITLE
Disabling GNOME's `at-spi` under LXQt by default?

### DISCUSSION THREAD

**Discussion by tsujan** [lxqt/lxqt#2453] [Ideas]
Its daemon starts as soon as some GTK-based apps are launched. If it only populated the system log with lines like `spi2-registr[583223]: AT-SPI: Cannot open default display`, I might not suggest this. But I think its disabling makes loading of folders containing many desktop files much faster.

It can be disabled by setting `NO_AT_BRIDGE` to `1` in LXQt Session Settings.

**Comment by yan12125** [comment]
I think at-spi is for accessibility services? I prefer to leave it as is. Those who don't need it can disable it by themselves, and if LXQt disables it by default, it may require some efforts to find out why it cannot be used for those who need it.

> `spi2-registr[583223]: AT-SPI: Cannot open default display`

Sounds like missing or invalid `$DISPLAY`. I don't have this issue on X11. Are you using Wayland?

**Reply by tsujan** [reply]
> Are you using Wayland?

No, I'm not.

As I said, if it was only because of those messages, I wouldn't suggest its disabling. I think it's a resource hog. Furthermore, why should a DE impose its daemon under other DEs?

**Reply by yan12125** [reply]
AT-SPI looks more like a cross-DE mechanism instead of GNOME-specific.  Qt also uses AT-SPI [1].

> I think it's a resource hog.

Not sure about this. I don't remember a case with high CPU usage or memory consumption from processes related to AT-SPI.

[1] https://doc.qt.io/qt-5/accessible.html

**Comment by isf63** [comment]
It is this specifically, https://en.wikipedia.org/wiki/Assistive_Technology_Service_Provider_Interface part of GNOME's 'ATK'

`NO_AT_BRIDGE=1` in LXQt Session disables at-spi2-registryd, but not at-spi-bus-launcher. at-spi-bus-launcher is launched in `/usr/lib/systemd/user/at-spi-dbus-bus.service` and `/etc/xdg/autostart/at-spi-dbus-bus.desktop`.

**Reply by yan12125** [reply]
Thanks for the information. I'm fine with `NO_AT_BRIDGE=1` if it doesn't affect functionality of assistive technologies.
